src/BingAdsEventForwarder.js

```diff
@@ -27,18 +27,52 @@ var MessageType = {
     Commerce: 16,
 };
 
+var bingConsentValues = { Denied: 'denied', Granted: 'granted' };
+var bingConsentProperties = ['ad_storage'];
+var bingToMpConsentSettingsMapping = {
+    ad_storage: 'defaultAdStorageConsentWeb',
+};
+
 var constructor = function() {
     var self = this;
     var isInitialized = false;
     var forwarderSettings = null;
     var reportingService = null;
 
+    self.consentMappings = [];
+    self.consentPayloadAsString = '';
+    self.consentPayloadDefaults = {};
+
     self.name = name;
 
     function initForwarder(settings, service, testMode) {
+        console.warn('BING Local DEV');
+
         forwarderSettings = settings;
         reportingService = service;
 
+        if (forwarderSettings.consentMapingWeb) {
+            self.consentMappings = parseSettingsString(
+                forwarderSettings.consentMapingWeb
+            );
+        }
+        self.consentPayloadDefaults = getConsentSettings(forwarderSettings);
+
+        var initialConsentPayload = cloneObject(self.consentPayloadDefaults);
+        var userConsentState = getUserConsentState();
+
+        var updatedConsentPayload = generateConsentPayload(
+            userConsentState,
+            self.consentMappings
+        );
+
+        var consentPayload;
+        if (!isEmpty(initialConsentPayload)) {
+            consentPayload = initialConsentPayload;
```

Comment:
under what conditions is there not an `initialConsentPayload`? `consentPayloadDefaults` will always have something in it because even if a customer has `Unspecified` selected, we would default to `Granted`, right?  And the MSFT docs say to always pass the default, so I don't think we should include a user consent state ever here, but instead send it _after_  defaults are set.  Just like the fix Mo had for GTM.